packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -725,3 +725,86 @@ class _MotionEventsDispatcher {
   bool isSinglePointerAction(PointerEvent event) =>
       !(event is PointerDownEvent) && !(event is PointerUpEvent);
 }
+
+/// A render object for embedding a platform view.
+///
+/// [PlatformViewRenderBox] presents an existing platform view, integrates it with the gesture arenas system
+/// and adds relevant semantic nodes to the semantics tree.
+/// It inserts a [PlatformViewLayer] to the layer tree.
+class PlatformViewRenderBox extends RenderBox {
+
+  /// Creating a render object for a [PlatformViewSurface].
+  ///
+  /// The `controller` and `id` must not be null.
+  PlatformViewRenderBox({
+    @required PlatformViewController controller,
+    @required int id,
+  }) : assert(controller != null),
+       assert(id != null),
+       _controller = controller,
+       _id = id;
+
+  /// The [PlatformViewController] associated with this render object.
+  PlatformViewController get controller => _controller;
+
+  /// Sets the [controller] for this render object.
+  ///
+  /// This value must not be null, and setting it to a new value will result in a repaint
+  set controller(PlatformViewController controller) {
+    assert(controller != null);
+    final bool needsPaint = _controller != controller;
+    _controller = controller;
+    if (needsPaint) {
+       markNeedsPaint();
+    }
+  }
+
+  /// The [id] of the platform view that is associated with this render object.
+  int get id => _id;
+
+  /// Sets the [id] of the platform view that is associated with this render object.
+  ///
+  /// This value must not be null and non-negative, and setting it to a new value will trigger a semantics update.
+  set id(int id) {
```

Comment:
Looking at this again I'm worried that there's not a single source of truth for the platform view that is associated with this render object (e.g the controller and the id may belong to a different view). What do you think about instead requiring that a PlatformViewController exposes its id and using it as the single source of truth? we could assert after creation that a PlatformViewController reports the id we assigned to it.

Reply:
done